src/examples/patient-deduplication.ts

```diff
@@ -0,0 +1,50 @@
+import { BotEvent, getDisplayString, getReferenceString, MedplumClient } from '@medplum/core';
+import { Patient } from '@medplum/fhirtypes';
+
+export async function handler(medplum: MedplumClient, event: BotEvent): Promise<any> {
+  //This bot should only be triggered by a Patient resource Subscription only
+  const patient = event.input as Patient;
+  if (patient.resourceType !== 'Patient') {
+    throw new Error('Unexpected input. Expected Patient.');
+  }
+
+  const identifier = patient.identifier?.[0].value?.toString();
+
+  const existingPatient = await medplum.searchOne('Patient', 'identifier=' + identifier);
```

Comment:
You probably want a filter to make sure you're searching for patients `!=` the input.

I.e. (untested)

```ts
medplum.searchOne('Patient', 'identifier=' + identifier + '&_id:not=' + patient.id);
```

Reply:
Bump this ^ - right now you're relying on `searchOne` not to return the input patient, which may or may not be true